Libs/DICOM/Core/ctkDICOMIndexer.cpp

```diff
@@ -318,3 +328,32 @@ void ctkDICOMIndexer::cancel()
   Q_D(ctkDICOMIndexer);
   d->Canceled = true;
 }
+
+//----------------------------------------------------------------------------
+void ctkDICOMIndexer::startIndexing(ctkDICOMDatabase& ctkDICOMDatabase)
```

Comment:
I suggest to create a `ctkScopedDICOMIndexing` object that would be allocated on the stack. On construction, it would call `startIndexing`, on destruction, it would call `endIndexing`.

Similar behavior to https://github.com/commontk/CTK/blob/master/Libs/Core/ctkScopedCurrentDir.cpp

Reply:
OK, I'll add a simple scoped guard class.